linalg/petsc.hpp

```diff
@@ -181,7 +181,8 @@ class PetscParMatrix : public Operator
        PETSc format @a tid. */
    /** The supported type ids are: Operator::PETSC_MATAIJ,
        Operator::PETSC_MATIS, and Operator::PETSC_MATSHELL. */
-   PetscParMatrix(const HypreParMatrix *ha, Operator::Type tid);
+   PetscParMatrix(const HypreParMatrix *ha,
+                  Operator::Type tid = Operator::PETSC_MATAIJ);
```

Comment:
[Posting my comment again since it was hidden by Tzanio's commit]:
I think we should add `explicit` to this constructor.

Reply:
@stefanozampini do you have any objections to label the constructor `explicit`?